plugins/utils/tmux-navigator.nix

```diff
@@ -0,0 +1,42 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+} @ attrs:
+with lib; let
+  cfg = config.plugins.tmux-navigator;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.tmux-navigator = {
+    enable = mkEnableOption "Enable Tmux-Navigator (see https://github.com/christoomey/vim-tmux-navigator for tmux installation instruction)";
+
+    package = helpers.mkPackageOption "tmux-navigator" pkgs.vimPlugins.tmux-navigator;
+
+    tmuxNavigatorSaveOnSwitch = helpers.mkNullOrOption lib.types.int ''
+      null: don't save on switch
+      1: update (write the current buffer, but only if changed)
+      2: wall (write all buffers)
+    '';
```

Comment:
If this is an enum couldn't you use a `helpers.mkNullOrOption (lib.types.enum [1 2])` to be more type aware ? Morever can you add to the description the upstream default value

Reply:
ah nice, didn't knew this type.